src/handlers/gestures/reanimatedWrapper.ts

```diff
@@ -34,7 +34,6 @@ try {
     // @ts-ignore Make sure the loaded module is actually Reanimated, if it's not
     // reset the module to undefined so we can fallback to the default implementation
     Reanimated = undefined;
-    throw new Error('react-native-reanimated is not found');
   }
 
   if (!Reanimated.setGestureState) {
```

Comment:
Without `throw new Error` here the execution will continue. In the previous line, we assigned `Reanimated = undefined` and in the following line we have a check `!Reanimated.setGestureState`. If `Reanimated` is undefined, this check will fail.

Reply:
Good thing you've caught it 😅. I've put the code that followed the throw into the else branch so it should be clearer now.

Reply:
Honestly, the current flow looks even more complicated. What about the following?

```ts
try {
  Reanimated = require('react-native-reanimated');
} catch (e) {
  // When 'react-native-reanimated' is not available we want to
  // quietly continue
}

if (!Reanimated?.useSharedValue) {
  // @ts-ignore Make sure the loaded module is actually Reanimated, if it's not
  // reset the module to undefined so we can fallback to the default implementation
  Reanimated = undefined;
}

if (!Reanimated?.setGestureState) {
  Reanimated.setGestureState = () => {
    'worklet';
    console.warn(
      tagMessage(
        'Please use newer version of react-native-reanimated in order to control state of the gestures.'
      )
    );
  };
}

export { Reanimated };
```

Reply:
What about something like this:

```jsx
try {
  Reanimated = require('react-native-reanimated');
} catch (e) {
  // When 'react-native-reanimated' is not available we want to quietly continue
  // @ts-ignore TS demands the variable to be initialized
  Reanimated = undefined;
}

if (!Reanimated?.useSharedValue) {
  // @ts-ignore Make sure the loaded module is actually Reanimated, if it's not
  // reset the module to undefined so we can fallback to the default implementation
  Reanimated = undefined;
} else if (!Reanimated?.setGestureState) {
  // The loaded module is Reanimated but it doesn't have the setGestureState defined
  Reanimated.setGestureState = () => {
    'worklet';
    console.warn(
      tagMessage(
        'Please use newer version of react-native-reanimated in order to control state of the gestures.'
      )
    );
  };
}

export { Reanimated };
```

In your snippet, the execution continues after the first `if` and we're back where we've started 😅.

Reply:
Also, this is a nice place to use IIFE and early returns.